src/schema.cpp

```diff
@@ -148,17 +148,32 @@ const char* MODELS_CONFIG_SCHEMA = R"({
 		},
 		"node_config": {
 			"type": "object",
-			"required": ["name", "model_name", "inputs", "outputs"],
+			"required": ["name", "type", "inputs", "outputs"],
+			"oneOf": [
+    			{
+        			"properties": { "type": { "enum": ["custom"] } },
+        			"required": ["library_name"],
+					"not": { "required": ["model_name"] }
```

Comment:
Shouldn't we have here:
"additionalProperties": false?

Reply:
I think adding "additionalProperties": false here means we don't want any properties except "type" and "library_name".